HW3/Task31/Queue.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Collections;
+
+namespace Task31
+{
+    public class Queue<T> : IQueue<T> where T : struct
+    {
```

Comment:
Not good idea to provide public access to elements of Queue, somebody can brake the structure, make it private and as a result refactor Tail() method)))